### PR TITLE
script bans

### PR DESCRIPTION
**Pull request by minrk**
- secrets/ban.py contains definitions for banned ips and domains
- running secrets/ban.py updates dns-based bans via kube-dns configmap
- running firewall-rules updates ip-based firewalls in ingress/egress

Running these scripts is *not* part of automatic Travis deployment (travis account doesn't have sufficient privileges), so it must be run locally at the moment.

### CODE REVIEW

**Comment by choldgraf** [comment]
is this something we should add to the SRE guide? (how to ban/unban IPs)?

**Comment by minrk** [comment]
> is this something we should add to the SRE guide?

yes!

**Comment by choldgraf** [comment]
@minrk is this ready to merge? I'm happy to open an issue for adding this to the docs so it isn't blocking on this. But probably needs a review too I think

File: scripts/firewall-rules [hunk 1]
```diff
@@ -68,10 +85,7 @@ def main(cluster, replace=False):
     # services_cidr = cluster_info['servicesIpv4Cidr']
     # kube_cidr = ','.join([cluster_cidr, services_cidr])
 
-    rules = {
-        rule['name']: rule
-        for rule in gcloud('compute', 'firewall-rules', 'list')
-    }
+    rules = {rule['name']: rule for rule in gcloud('compute', 'firewall-rules', 'list')}
 
     # get the node network tags for our cluster
```
[the remark this answers is not in the document]
Comment (willingc) [reply scripts/firewall-rules]: If I run:
```
NAME                           ZONE           MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP      STATUS
gke-prod-a-hm32-d85259ee-1qm7  us-central1-a  n1-highmem-32               10.128.0.3   35.224.219.184   RUNNING
gke-prod-a-hm32-d85259ee-6dld  us-central1-a  n1-highmem-32               10.128.0.2   23.251.147.76    RUNNING
gke-prod-a-hm32-d85259ee-xgrr  us-central1-a  n1-highmem-32               10.128.0.4   104.198.227.176  RUNNING
ERROR: (gcloud.compute.instances.list) Some requests did not succeed:
 - Invalid value for field 'zone': 'europe-north1-a'. Unknown zone.
 - Invalid value for field 'zone': 'europe-north1-b'. Unknown zone.
 - Invalid value for field 'zone': 'europe-north1-c'. Unknown zone.
```
Checking my understanding...In this example:
- nodes would contain 3 instances
- instance group prefix would be equal to the NAME in the chart

File: scripts/firewall-rules [hunk 2]
```diff
@@ -84,10 +98,9 @@ def main(cluster, replace=False):
 
     # get nodes for our cluster
```
Comment (willingc) [inline scripts/firewall-rules]:
```
# get nodes in our cluster, if any,  that begin with one of the instance_group_prefixes
```

File: scripts/firewall-rules [hunk 3]
```diff
@@ -68,10 +85,7 @@ def main(cluster, replace=False):
     # services_cidr = cluster_info['servicesIpv4Cidr']
     # kube_cidr = ','.join([cluster_cidr, services_cidr])
 
-    rules = {
-        rule['name']: rule
-        for rule in gcloud('compute', 'firewall-rules', 'list')
-    }
+    rules = {rule['name']: rule for rule in gcloud('compute', 'firewall-rules', 'list')}
```
Comment (willingc) [inline scripts/firewall-rules]:
```
# get a dictionary of firewall rules
```

File: scripts/firewall-rules [hunk 4]
```diff
@@ -98,54 +111,102 @@ def main(cluster, replace=False):
         node_tags.update(node['tags']['items'])
     assert node_tags, f"No node tags for {instance_group_prefixes}"
 
-    base_rule = {
-        'direction': 'egress',
-        'source-tags': ','.join(node_tags)
-    }
+    # Banned ips
+
```
Comment (willingc) [inline scripts/firewall-rules]:
```
# Add ingress and egress rules for banned ips
```

**Comment by minrk** [comment]
Added commands to the docs and comments to firewall-rules. Should be set now.

**Comment by willingc** [comment]
Thanks @minrk 👍

**Comment by choldgraf** [comment]
woo thanks!!

**Pull request merged**